### PR TITLE
Fix the blob for s3 prefixed key 

### PR DESCRIPTION
**Pull request by khrm**
# Changes
Prefix behave differently than normal folders. Folders don't
require a leading "/" but Prefix does.
Also, refactor to avoid the error during the clean func call.

Also, legacy APIs should only be supported for v1beta1 or using LogType record.
This adds it to a v1beta1 Type when legacy param is supplied.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
If using prefix bucket, please pass // in LOGS_PATH, e.g.: LOGS_PATH=//logs
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/assign @enarha

**Review by khrm — COMMENTED** [review]
/kind bug

File: pkg/api/server/v1alpha2/plugin/plugin_logs.go [hunk 1]
```diff
@@ -304,10 +306,9 @@ func getBlobLogs(s *LogServer, writer io.Writer, parent string, rec *db.Record)
 		err := errors.New("pipelinerun not supported, please use taskrun")
 		s.logger.Error(err)
 		return err
-	case typeTaskRun:
-		logRec := &db.Record{}
-		if legacy {
-			logRec, err = getLogRecord(s.db, parent, rec.ResultID, rec.Name)
+	case typeTaskRunV1Beta1:
+		if legacy && rec.Type == typeTaskRunV1Beta1 {
```
Comment (enarha) [inline pkg/api/server/v1alpha2/plugin/plugin_logs.go:310]: Is `&& rec.Type == typeTaskRunV1Beta1` really required? Isn't that ensured by `case typeTaskRunV1Beta1`?

Comment (khrm) [reply pkg/api/server/v1alpha2/plugin/plugin_logs.go:310]: It's not required, I forgot to remove this.

Comment (khrm) [reply pkg/api/server/v1alpha2/plugin/plugin_logs.go]: Fixed this.

**Comment by khrm** [comment]
@enarha Please review this again.

**Pull request merged**